zlib-rs/src/weak_slice.rs

```diff
@@ -59,3 +63,30 @@ impl<'a, T> WeakSliceMut<'a, T> {
         }
     }
 }
+
+#[derive(Debug)]
+pub(crate) struct WeakArrayMut<'a, T, const N: usize> {
+    ptr: *mut [T; N],
+    _marker: PhantomData<&'a mut [T; N]>,
+}
+
+impl<'a, T, const N: usize> WeakArrayMut<'a, T, N> {
+    pub(crate) unsafe fn from_ptr(ptr: *mut [T; N]) -> Self {
+        Self {
+            ptr,
+            _marker: PhantomData,
+        }
+    }
```

Comment:
Should document the preconditions of this function, as was done in `WeakSliceMut`. (i.e. the pointed to slice needs to satisfy the requirements of from_raw_parts()`.